src/Core/Scripting/OOJSShip.m

```diff
@@ -2216,7 +2216,7 @@ static JSBool ShipAddCargoEntity(JSContext *context, uintN argc, jsval *vp)
 				   !JSVAL_IS_OBJECT(OOJS_ARGV[0]) ||
 				   !JSShipGetShipEntity(context, JSVAL_TO_OBJECT(OOJS_ARGV[0]), &target)))
 	{
```

Comment:
A check against NULL JS value is needed here, otherwise something as simple as player.ship.addCargoEntity(player.ship.target, true, true) will throw a native exception.